warehouse/manage/views.py

```diff
@@ -559,7 +559,24 @@ def __init__(self, request):
 
     @property
     def project_names(self):
-        projects = user_projects(self.request)["projects_owned"]
+        projects_owned_or_maintained = (
+            self.request.db.query(Project.id)
+            .join(Role.project)
+            .filter(
+                Role.role_name.in_(("Owner", "Maintainer")),
+                Role.user == self.request.user,
+            )
+            .subquery()
+        )
+        projects = (
+            self.request.db.query(Project)
+            .join(
+                projects_owned_or_maintained,
+                Project.id == projects_owned_or_maintained.c.id,
+            )
+            .order_by(Project.name)
+            .all()
+        )
         return [project.name for project in projects]
```

Comment:
Looks like this all could probably just become:

```python
sorted(project.name for project in request.user.projects)
```

unless I'm missing something.